### DISCUSSION TITLE
How can I use it with yew?

### DISCUSSION THREAD

**Discussion by Raduc4** [devashishdxt/rexie#28] [Q&A]
Hi, very happy that I found this project, my question is how can I implement it into my yew frontend app?
The entire app is sync and can't make it async.
Can you advise something?

**Comment by devashishdxt** [comment]
You can use [`wasm_bindgen_futures::spawn_local`](https://docs.rs/wasm-bindgen-futures/0.4.33/wasm_bindgen_futures/fn.spawn_local.html).

**Comment by Raduc4** [comment]
Thank you for the reply.
Unfortunately I'm getting an error.
```
type mismatch resolving `<impl std::future::Future<Output = std::result::Result<rexie::Rexie, rexie::Error>> as std::future::Future>::Output == ()`
expected unit type `()`
        found enum `std::result::Result<rexie::Rexie, rexie::Error>`rustc[E0271](https://doc.rust-lang.org/error-index.html#E0271)
lib.rs(79, 15): required by a bound in `spawn_local`
```

**Comment by devashishdxt** [comment]
I cannot help with this error without looking at the code. Looks like you need to handle errors.

**Comment by Raduc4** [comment]
Here is the line, it looks like that wasm_bindgen_futures::spawn local needs a Future<Output = ()> as argument but my future gives Future<Output  = Rexie>
[The code ](https://github.com/Raduc4/evaluation-digitalization/blob/6dd1847ca844ca22104b31d1544142246d57ffd0/src/pages/home.rs#L55-L57)

**Comment by devashishdxt** [comment]
I think you're returning `Rexie` from build database. Maybe try and store it in a `UseStateHandle`. You cannot return values from spawned futures.

**Comment by ctm** [comment]
I realize this is a bit late, but as [I mention](https://github.com/devashishdxt/rexie/issues/23#issuecomment-2375438018) in a different issue, I have a working demo using Yew that I don't mind sharing with people.
